레스토랑 조회 구현하기

kokoball opened the pull request:
안녕하세요 트레이너님!!

5주차 과제 첫 PR 입니다!!!

배운 TDD방식으로 코딩하려고 노력했습니다.

우선 카테고리 관련된 작업을 먼저 했습니다 완료했습니다!!

감사합니다😊😊

kokoball committed:
App test 작성

kokoball committed:
App 구현

kokoball committed:
Categories test 추가

kokoball committed:
Categories 구현

kokoball committed:
CategoriesContainer test 및 구현

kokoball committed:
CategoriesContainer APP에 연결

kokoball committed:
reducer test 작성

kokoball committed:
App 에 dispatch 추가하기

kokoball committed:
비동기 액션 및 Redux Thunk 추가

moonkii commented:
![image](https://user-images.githubusercontent.com/46776329/132725361-998876bf-9ba1-4301-9cd9-b9ccea0d4889.png)

TDD 싸이클도 잘 지켜서 하셨고 코드도 잘 리팩터링하신 것 같아요! 👍 
전체적으로 잘 하셨는데 카테고리만 만들어서 리팩터링하신게 아쉬워요!
처음하실 때 App 컴포넌트에 지역 버튼까지해서 완성된 형태에서 리팩터링을 고민해보고 진행하셨으면
완벽했을 것 같아요. 💯

kokoball committed:
Categories.test 수정 (button 추가)

kokoball committed:
Categories button 형태 추가(수정)

radio button을 이용해 그룹에 한 개만 클릭되게 만들었습니다.
다만 cllick 시 text 값을 수정하는 방법을 해결하지 못했고,
input 형태를 지울려면 style를 수정해야하는지 방법을 찾고 있습니다.

kokoball committed:
Regions test 추가

kokoball committed:
Regions 구현

kokoball committed:
Regions 리펙토링(fixtures/regions 생성)

kokoball committed:
GionsContainer test 및 구현

kokoball committed:
reducer Regions test 작성

kokoball committed:
reducer setRegions 추가

moonkii commented on src/Categories.jsx:
```diff
@@ -0,0 +1,17 @@
+  return (
+    <ul>
+      {categories.map((category) => (
+        <li>
+          <input key={category.id} type="radio" onClick={handleInputText} name="categories" />
+          <label htmlFor={category.id}>{category.name}</label>
+        </li>
```
이부분은 하나의 category 로 리팩터링 해보면 어떨까요?
그럼 관심사가 더 세세하게 나눠질 수 있을 것 같아요.

moonkii commented:
지금 구조에서는 부분부분들의 구조를 미리 결정해서 만들고 있어서 프로그램이 복잡해질수록 구조잡기가 어려워질 수 있어요. 전체를 완성한 다음 컴포넌트를 분리하는 구조로 가야 의미상으로 구분되는 지점들과 유연한 구조를 찾는게 수월하실 거에요.

kokoball committed:
Finish Restaurant-App
